packages/manager/modules/pci/src/projects/project/data-processing/submit-job/spark-config/spark-config.controller.js

```diff
@@ -77,6 +107,22 @@ export default class {
       .listObjects(this.projectId, containerId)
       .then((container) => {
         this.containerObjects = container.objects;
+        // try to detect correct job type and file
+        const pythonFiles = container.objects.filter((object) =>
+          PYTHON_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        const javaFiles = container.objects.filter((object) =>
+          JAVA_FILES_CONTENT_TYPES.includes(object.contentType),
+        );
+        if (pythonFiles.length > 0) {
+          this.state.jobType = 'python';
+          this.state.mainApplicationCode = pythonFiles[0].name;
+        } else if (javaFiles.length > 0) {
+          this.state.jobType = 'java';
```

Comment:
why not using `JOB_TYPE` constant here?

Reply:
Done here: https://github.com/ovh/manager/pull/7069/commits/67f807ec48058a5f5652d27e617e21166749fd46